Packages/com.unity.inputsystem/InputSystem/Plugins/DualShock/DualShockGamepadHID.cs

```diff
@@ -549,6 +552,50 @@ unsafe bool IEventPreProcessor.PreProcessEvent(InputEventPtr eventPtr)
                 return false; // skip unrecognized reportId
         }
 
+        public void OnNextUpdate()
+        {
+        }
+
+        // filter out three lower bits as jitter noise
+        internal const byte AxisDeadZoneLow = 0b01111000;
+        internal const byte AxisDeadZoneHigh = 0b10000111;
+
+        public unsafe void OnStateEvent(InputEventPtr eventPtr)
+        {
+            if (eventPtr.type == StateEvent.Type && eventPtr.stateFormat == DualSenseHIDInputReport.Format)
+            {
+                var currentState = (DualSenseHIDInputReport*)((byte*)currentStatePtr + m_StateBlock.byteOffset);
+                var newState = (DualSenseHIDInputReport*)StateEvent.FromUnchecked(eventPtr)->state;
+
+                var actuated =
+                    // we need to make device current if axes are outside of deadzone specifying hardware jitter of sticks around zero point
+                    newState->leftStickX<AxisDeadZoneLow
+                                         || newState->leftStickX> AxisDeadZoneHigh
+                    || newState->leftStickY<AxisDeadZoneLow
+                                            || newState->leftStickY> AxisDeadZoneHigh
+                    || newState->rightStickX<AxisDeadZoneLow
+                                             || newState->rightStickX> AxisDeadZoneHigh
+                    || newState->rightStickY<AxisDeadZoneLow
+                                             || newState->rightStickY> AxisDeadZoneHigh
+                    // we need to make device current if triggers or buttons state change
+                    || newState->leftTrigger != currentState->leftTrigger
+                    || newState->rightTrigger != currentState->rightTrigger
+                    || newState->buttons0 != currentState->buttons0
+                    || newState->buttons1 != currentState->buttons1
+                    || newState->buttons2 != currentState->buttons2;
```

Comment:
Do we actually need to check all these non-jittery controls? Could we assume that because there's a state event we always want to make this device current except if the sticks have jittered? Or in other words, set actuated to true by default and only false if there's stick movement in the deadzones.

Reply:
We have to check every event because these gamepads send an event for every sensor update, which is around 100-300 Hz, but because we don't process sensors yet we get bunch of events where nothing changed.

Reply:
Ah of course. Missed that.